src/parachain/amm/encoding.ts

```diff
@@ -0,0 +1,76 @@
+import { MonetaryAmount } from "@interlay/monetary-js";
+import { ApiPromise } from "@polkadot/api";
+import {
+    Trade,
+    CurrencyExt,
+    InterbtcPrimitivesCurrencyId,
+    isStableMultiPathElement,
+    isStableMetaMultiPathElement,
+    newCurrencyId,
+} from "../..";
+
+function encodeSwapParamsForStandardPoolsOnly(
+    api: ApiPromise,
+    trade: Trade,
+    minimumAmountOut: MonetaryAmount<CurrencyExt>
+): {
+    amountIn: string;
+    amountOutMin: string;
+    path: Array<InterbtcPrimitivesCurrencyId>;
+} {
+    const amountIn = trade.inputAmount._rawAmount.toString();
+    const amountOutMin = minimumAmountOut._rawAmount.toString();
+    const path = trade.path
+        .reduce((result, currentPathElement) => [...result, currentPathElement.output], [trade.path[0].input])
+        .map((currency) => newCurrencyId(api, currency));
+    return { amountIn, amountOutMin, path };
+}
+
+function encodeSwapParamsForStandardAndStablePools(
+    api: ApiPromise,
+    trade: Trade,
+    minimumAmountOut: MonetaryAmount<CurrencyExt>
+): {
+    amountIn: string;
+    amountOutMin: string;
+    path: (
+        | {
+              Stable: {
+                  poolId: number;
+                  basePoolId: number | false;
+                  mode: string;
+                  fromCurrency: InterbtcPrimitivesCurrencyId;
+                  toCurrency: InterbtcPrimitivesCurrencyId;
+              };
+          }
+        | {
+              Normal: Array<InterbtcPrimitivesCurrencyId>;
+          }
+    )[];
+} {
+    const amountIn = trade.inputAmount._rawAmount.toString();
+    const amountOutMin = minimumAmountOut._rawAmount.toString();
+    const path = trade.path.map((pathElement) =>
+        isStableMultiPathElement(pathElement)
+            ? {
+                  Stable: {
+                      poolId: pathElement.pool.poolId,
+                      basePoolId: isStableMetaMultiPathElement(pathElement) && pathElement.basePool.poolId,
```

Comment:
What happens if `isStableMetaMultiPathElement()` returns false? 
Is any falsy value valid for `basePoolId`?

Reply:
I think this was mistake, good catch! Zenlink sdk returns undefined here, but now I changed it to explicit 0.